dcc-release-job/dcc-release-job-id/src/main/java/org/icgc/dcc/release/job/id/parser/ExportStringParser.java

```diff
@@ -0,0 +1,27 @@
+package org.icgc.dcc.release.job.id.parser;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.icgc.dcc.common.core.util.Separators;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Created by gguo on 6/8/17.
+ */
+public class ExportStringParser<IDTYPE> {
+
+    public Map<IDTYPE, String> parse(String str, GenerateIDTypeInstance<IDTYPE> generator){
+        return
+            Separators.getCorrespondingSplitter(Separators.NEWLINE).splitToList(str).stream().map(
```

Comment:
Something like this would be much easier to read.
```java
import static org.icgc.dcc.common.core.util.Splitters.NEWLINE;
import static org.icgc.dcc.common.core.util.Splitters.TAB;
import static org.icgc.dcc.common.core.util.stream.Collectors.toImmutableMap;
.
.
.

  public Map<IDTYPE, String> parse1(String str, GenerateIDTypeInstance<IDTYPE> generator) {
    return NEWLINE.splitToList(str).stream()
        .map(TAB::splitToList)
        .map(generator::generate)
        .collect(toImmutableMap(Entry::getKey, Entry::getValue));
  }

  public interface GenerateIDTypeInstance<IDTYPE> {
    Entry<IDTYPE, String> generate(List<String> fields);
  }
```

You can probably get away with using `Entry` and not have to bring in any apache commons. 
We also have a useful `dcc-common` project where we have a ton of useful things to use like splitters and collectors.